feature/experimental/qos/otg_tests/two_sp_queue_traffic_test/two_sp_queue_traffic_test.go

```diff
@@ -0,0 +1,1586 @@
+	}, {
+		desc:         "Non-oversubscription AF4 and AF3 traffic",
+		trafficFlows: nonOversubscribedTrafficFlows7,
+	}, {
+		desc:         "Non-oversubscription AF4 and AF2 traffic",
+		trafficFlows: nonOversubscribedTrafficFlows8,
+	}, {
+		desc:         "Non-oversubscription AF4 and AF1 traffic",
+		trafficFlows: nonOversubscribedTrafficFlows9,
+	}, {
+		desc:         "Non-oversubscription AF4 and BE0 traffic",
+		trafficFlows: nonOversubscribedTrafficFlows10,
+	}, {
+		desc:         "Non-oversubscription AF4 and BE1 traffic",
+		trafficFlows: nonOversubscribedTrafficFlows11,
+	}, {
+		desc:         "Oversubscription NC1 and AF4 traffic with half AF4 dropped",
+		trafficFlows: oversubscribedTrafficFlows1,
+	}, {
+		desc:         "Oversubscription NC1 and AF3 traffic with half AF3 dropped",
+		trafficFlows: oversubscribedTrafficFlows2,
+	}, {
+		desc:         "Oversubscription NC1 and AF2 traffic with half AF2 dropped",
+		trafficFlows: oversubscribedTrafficFlows3,
+	}, {
+		desc:         "Oversubscription NC1 and AF1 traffic with half AF1 dropped",
+		trafficFlows: oversubscribedTrafficFlows4,
+	}, {
+		desc:         "Oversubscription NC1 and BE0 traffic with half BE0 dropped",
+		trafficFlows: oversubscribedTrafficFlows5,
+	}, {
+		desc:         "Oversubscription NC1 and BE1 traffic with half BE1 dropped",
+		trafficFlows: oversubscribedTrafficFlows6,
+	}, {
+		desc:         "Oversubscription AF4 and AF3 traffic with half AF3 dropped",
+		trafficFlows: oversubscribedTrafficFlows7,
+	}, {
+		desc:         "Oversubscription AF4 and AF2 traffic with half AF2 dropped",
+		trafficFlows: oversubscribedTrafficFlows8,
+	}, {
+		desc:         "Oversubscription AF4 and AF1 traffic with half AF1 dropped",
+		trafficFlows: oversubscribedTrafficFlows9,
+	}, {
+		desc:         "Oversubscription AF4 and BE0 traffic with half BE0 dropped",
+		trafficFlows: oversubscribedTrafficFlows10,
+	}, {
+		desc:         "Oversubscription AF4 and BE1 traffic with half BE1 dropped",
+		trafficFlows: oversubscribedTrafficFlows11,
+	}}
+
+	for _, tc := range cases {
+		t.Run(tc.desc, func(t *testing.T) {
+			trafficFlows := tc.trafficFlows
+			top.Flows().Clear()
+
+			for trafficID, data := range trafficFlows {
+				t.Logf("Configuring flow %s", trafficID)
+				flow := top.Flows().Add().SetName(trafficID)
+				flow.Metrics().SetEnable(true)
+				flow.TxRx().Device().SetTxNames([]string{data.inputIntf.Name + ".IPv4"}).SetRxNames([]string{intf3.Name + ".IPv4"})
+				ethHeader := flow.Packet().Add().Ethernet()
+				ethHeader.Src().SetValue(data.inputIntf.MAC)
+
+				ipHeader := flow.Packet().Add().Ipv4()
+				ipHeader.Src().SetValue(data.inputIntf.IPv4)
+				ipHeader.Dst().SetValue(intf3.IPv4)
+				ipHeader.Priority().Dscp().Phb().SetValue(int32(data.dscp))
+
+				flow.Size().SetFixed(int32(data.frameSize))
+				flow.Rate().SetPercentage(float32(data.trafficRate))
+			}
+
+			ate.OTG().PushConfig(t, top)
+			ate.OTG().StartProtocols(t)
+			time.Sleep(30 * time.Second)
```

Comment:
Is there something else we can wait on here? (Eg. IP neighbors maybe?)

Reply:
Added WaitforArp